clientServiceClient/api.go

```diff
@@ -0,0 +1,52 @@
+package clientServiceClient
+
+import (
+	"github.com/google/uuid"
+)
+
+type AdminListRequest struct {
+	NextToken int64 `json:"next_token"`
+	Limit     int   `json:"limit"`
+}
+
+type AdminListResponse struct {
+	Clients   []Client `json:"clients"`
+	NextToken int64    `json:"next_token"`
+}
+
+type AdminGetResponse struct {
+	Client
+}
+
+type ClientGetResponse struct {
+	Client
+}
+
+// ClientRegisterRequest captures the information sent to create a client.
+type ClientRegisterRequest struct {
+	RegistrationToken string `json:"token"`
+	Client            struct {
```

Comment:
why this inline struct versus using the client type on line 43 below?

Reply:
This inline struct doesn't accept these as fields

	ClientID    uuid.UUID         `json:"client_id"`
	APIKeyID    string            `json:"api_key_id"`
	Name        string            `json:"name"`
	Enabled     bool              `json:"enabled"`
	Meta        map[string]string `json:"meta,omitempty"`

Reply:
Right, but why do we need a separate struct versus having the endpoint that processes this request only extract the fields of interest?

Reply:
To provide the clarity of a specific type, what are we gaining by using the separate struct with additional information except for saving lines? I agree this inline struct is a little ugly, but the fields the user provides will actually be used.

Reply:
As long as we don't need this representation of a client elsewhere, my answer to your question is: `nothing`.